src/lein_monolith/plugin.clj

```diff
@@ -154,17 +176,17 @@
 
 (defn middleware
   "Handles inherited properties in monolith subprojects by looking for the
-  `:monolith/inherit` key."
+  `:monolith/inherit` key, or sets the managed dependencies if :monolith/dependency-set is set."
   ([project]
    (middleware project nil))
   ([project monolith]
-   (if (or (not (:monolith/inherit project))
+   (if (or (not-any? #{:monolith/inherit :monolith/dependency-set} (keys project))
            (:monolith/active (meta project)))
```

Comment:
This also would need to check the other inheritance keys for correct behavior. At this point, I think it's safer to only check `:monolith/active` here, and then just check whether the generated profiles map is empty to determine whether we need to add/merge into the project.